zuul-core/src/test/java/com/netflix/netty/common/proxyprotocol/StripUntrustedProxyHeadersHandlerTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.netty.common.proxyprotocol;
+
+import static com.netflix.zuul.netty.server.ssl.SslHandshakeInfoHandler.ATTR_SSL_INFO;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import com.google.common.collect.ImmutableList;
+import com.netflix.netty.common.proxyprotocol.StripUntrustedProxyHeadersHandler.AllowWhen;
+import com.netflix.netty.common.ssl.SslHandshakeInfo;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.util.AttributeKey;
+import io.netty.util.DefaultAttributeMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * Strip Untrusted Proxy Headers Handler Test
+ *
+ * @author Arthur Gonigberg
+ * @since May 27, 2020
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class StripUntrustedProxyHeadersHandlerTest {
+
+    @Mock
+    private ChannelHandlerContext channelHandlerContext;
+    @Mock
+    private HttpRequest msg;
+    @Mock
+    private HttpHeaders headers;
+    @Mock
+    private Channel channel;
+    @Mock
+    private SslHandshakeInfo sslHandshakeInfo;
+
+
+    @Before
+    public void before() {
+        when(channelHandlerContext.channel()).thenReturn(channel);
+
+        DefaultAttributeMap attributeMap = new DefaultAttributeMap();
+        attributeMap.attr(ATTR_SSL_INFO).set(sslHandshakeInfo);
+        when(channel.attr(any())).thenAnswer(arg -> attributeMap.attr((AttributeKey) arg.getArguments()[0]));
+
+        when(msg.headers()).thenReturn(headers);
+        when(headers.get(eq(HttpHeaderNames.HOST))).thenReturn("netflix.com");
+    }
+
+    @Test
+    public void allow_never() throws Exception {
+        StripUntrustedProxyHeadersHandler stripHandler = getHandler(AllowWhen.NEVER);
+
+        stripHandler.channelRead(channelHandlerContext, msg);
+
+        verify(stripHandler).stripXFFHeaders(any());
```

Comment:
wouldn't it be more accurate to test that the headers are removed?

Reply:
In a sense yes, but maybe not for this specific unit. I could another test that validates that we're stripping headers properly.